### PR TITLE
fix(vararg): fix bug for local objects called

### PR DESCRIPTION
**Pull request by baudoliver7**
We fix here bug about readonly attribute when calling a local object with vararg as free attribute.
We added a puzzle to fix the case of external object called with vararg as free attribute.

Ref: #534

### CODE REVIEW

**Comment by baudoliver7** [comment]
@yegor256 please, check/merge

**Comment by baudoliver7** [comment]
@yegor256 please, check/merge

File: eo-maven-plugin/src/main/resources/org/eolang/maven/pre/called-objects.xsl [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+-->
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" id="pre-called-objects" version="2.0">
```
Comment (yegor256) [inline eo-maven-plugin/src/main/resources/org/eolang/maven/pre/called-objects.xsl:25]: @baudoliver7 could you please put a brief explanation of your operations here, in form of a comment. See how it's done here, for example: https://github.com/cqfn/eo/blob/master/eo-parser/src/main/resources/org/eolang/parser/vars-float-up.xsl#L26-L43

Comment (yegor256) [inline eo-maven-plugin/src/main/resources/org/eolang/maven/pre/called-objects.xsl:25]: @baudoliver7 the `@id` attribute is wrong. It should be the name of your XSL, see how it's done for other XSLs

**Comment by yegor256** [comment]
@baudoliver7 seems to be a good PR, thanks! Could you please add a comment, as I suggested above?

**Comment by baudoliver7** [comment]
@yegor256 Done as you suggested

**Comment by yegor256** [comment]
@baudoliver7 I don't think this will work when we don't have the source of the object. Exactly to solve this problem we have `AtVararg`, which accepts arguments one by one and wraps them into an array. What you are doing will work only for limited cases and will never work on a large scale.

**Comment by yegor256** [comment]
@baudoliver7 the following assumption is not correct. This code:

```
    [] > test
      [] > app
        (f 1 2 3) > @
      [args...] > f
        args.length > @
```

is NOT the same as this one:

```
    [] > test
      [] > app
        (f (* 1 2 3)) > @
      [args...] > f
        args.length > @
```

The object `f` doesn't expect an array of arrays, but you will give it exactly that: the array of arrays:

```
args = [(* 1 2 3)]
```

**Comment by yegor256** [comment]
@baudoliver7 I added an extra test at 25d0fbe8 Does it work for your branch?

**Comment by baudoliver7** [comment]
> @baudoliver7 the following assumption is not correct. This code:
> 
> ```
>     [] > test
>       [] > app
>         (f 1 2 3) > @
>       [args...] > f
>         args.length > @
> ```
> 
> is NOT the same as this one:
> 
> ```
>     [] > test
>       [] > app
>         (f (* 1 2 3)) > @
>       [args...] > f
>         args.length > @
> ```
> 
> The object `f` doesn't expect an array of arrays, but you will give it exactly that: the array of arrays:
> 
> ```
> args = [(* 1 2 3)]
> ```

You are right. But, I actually use an `unvaring` array instead of a simple array. In this way, `args` is not considered as array of arrays.

**Comment by baudoliver7** [comment]
> @baudoliver7 I don't think this will work when we don't have the source of the object we copy. Exactly to solve this problem we have `AtVararg`, which accepts arguments one by one and wraps them into an array. What you are doing will work only for limited cases and will never work on a large scale.

@yegor256 I agree with you. That's why I added a puzzle to solve it. About `AtVararg`, how could we use it with list arguments while we do not yet know the values to pass ?

**Comment by yegor256** [comment]
@baudoliver7 the puzzle you added won't be solved ever, and we will have to roll back your changes :(

**Comment by baudoliver7** [comment]
> @baudoliver7 the puzzle you added won't be solved ever, and we will have to roll back your changes :(

@yegor256 Sorry, I suggested a solution that resolved it in PR #573 and that works well. It's to avoid a big PR that I chose to close it and split my solution in small ones.

But, if this solution (describes in PR #573) doesn't suit you, I am open to your suggestions :)

You talked about passing arguments one by one to `AtVararg`. But, how could we use it while we do not yet know the values to pass ?

**Comment by baudoliver7** [comment]
@yegor256 Besides, regardless of whether the next puzzle can be solved or not, do you agree with me that this PR passes the tests ?
So why should we revert the changes if the next puzzle was not solvable ? (despite I think it is solvable :-) )

**Comment by yegor256** [comment]
@baudoliver7 this is fixed already, thanks for your contribution!

**Pull request closed without merging**